src/js/client/editprefset-prefSettingAdjuster-widget.js

```diff
@@ -0,0 +1,222 @@
+                    }
+                },
+                modelToDom: {
+                    "": {
+                        transform: {
+                            type: "gpii.devpmt.prefSettingAdjuster.valueInputToBinder",
+                            inputPath: ""
+                        }
+                    }
+                }
+            }
+        }
+    },
+    templates: {
+        initial: "editprefset-prefSettingAdjuster-widget"
+    },
+    invokers: {
+        renderInitialMarkup: {
+            func: "{that}.renderMarkup",
+            args: ["initial", "{that}.options.templates.initial", "{that}.model"]
+        },
+        saveUpdateValue: {
+            funcName: "gpii.devpmt.saveUpdateValue",
+            args: ["{that}", "{editPrefs}"]
+        },
+        watchInputKeys: {
+            funcName: "gpii.devpmt.watchInputKeys",
+            args: ["{that}", "{arguments}.0"]
+        },
+        updateBlankDisabling: {
+            funcName: "gpii.devpmt.updateBlankDisabling",
+            args: ["{that}"]
+        }
+    },
+    listeners: {
+        onMarkupRendered: [{
+            "this": "{that}.dom.okButton",
+            "method": "click",
+            args: ["{that}.saveUpdateValue"]
+        },
+        {
+            "this": "{that}.dom.valueInput",
+            "method": "keypress",
+            args: ["{that}.watchInputKeys"]
+        },
+        {
+            funcName: "fluid.focus",
+            args: ["{that}.dom.valueInput"]
+        },
+        {
+            func: "{that}.updateBlankDisabling"
+        }]
+    },
+    modelListeners: {
+        "current.blank": {
+            func: "{that}.updateBlankDisabling"
+        }
+    }
+});
+
+/**
+ * Updates the UI based on whether Blank is selected.
+ * If blank is selected, then the rest of the editing
+ * apparatus is disabled.
+ */
+gpii.devpmt.updateBlankDisabling = function (that) {
+    var disable = that.model.current.blank;
+    that.dom.locate("valueInput").prop("disabled", disable);
+};
+
+/**
+ * watchInputKeys - Watches value input and saves
+ * on Enter.
+ *
+ * @param (Object) that - prefSettingAdjuster
+ * @param (Object) e - jQuery Event
+ */
+gpii.devpmt.watchInputKeys = function (that, e) {
+    // Save the value when the enter key is pressed
+    if (e.keyCode === 13) {
+        that.saveUpdateValue();
+    };
+};
+
+/**
+ * Save/Update using the value the user has input. This is
+ * typically caused by clicking the Ok button.
+ */
+gpii.devpmt.saveUpdateValue = function (that, devpmt) {
+    that.applier.change("active", false);
+    var segs = ["contexts", that.model.current.context, "preferences"];
+    if (that.model.current.product) {
+        segs.push(that.model.current.product.replace(/\./g, "\\."), that.model.current.term.replace(/\./g, "\\."));
+    }
+    else {
+        segs.push(that.model.current.term.replace(/\./g, "\\."));
+    }
+    var path = "flatPrefs";
+    fluid.each(segs, function (item) { path += "." + item; });
```

Comment:
Why not just var path = ["flatPrefs"].concat(segs);

Reply:
this has been simplified to `var path = ["flatPrefs"].concat(segs).join(".");`